lib/apiservers/engine/backends/archive.go

```diff
@@ -461,20 +461,32 @@ func (wm *ArchiveStreamWriterMap) WriterForAsset(proxy proxy.VicArchiveProxy, ci
 }
 
 // Close visits all the archive writer in the trie and closes the actual io.WritCloser
-func (wm *ArchiveStreamWriterMap) Close() {
+func (wm *ArchiveStreamWriterMap) Close() error {
 	defer trace.End(trace.Begin(""))
 
+	numWriter := 0
 	closeStream := func(prefix patricia.Prefix, item patricia.Item) error {
 		if aw, ok := item.(*ArchiveWriter); ok && aw.writer != nil {
-			aw.writer.Close()
 			aw.tarWriter.Close()
-			aw.writer = nil
+			aw.writer.Close()
 			aw.tarWriter = nil
+			aw.writer = nil
+			numWriter++
 		}
 		return nil
 	}
 
 	wm.prefixTrie.Visit(closeStream)
+
+	var err error
+	// wait for all the streams to finish
+	for i := 0; i < numWriter; i++ {
```

Comment:
I am waiting for an arbitrary number of go routines to finish, so it's more for a state checking. Also I won't know how many of them returns errors.  I've looked into sync.waitGroup and using an additional array to store errors but it looks messier than just using the chan error. 

I can't use for range because the buffer allocated for channel can be larger than the number of writers I am waiting for. I have to allocate the buffer size before I know how many active writers are there. But yes, I will add a close after waiting for the writer.